uint16 requiredMaps
---
string mapJS